src/AppInstallerCommonCore/Settings.cpp

```diff
@@ -347,7 +350,132 @@ namespace AppInstaller::Settings
+                    }
+
+                    if (!tempSettingsTrusted)
+                    {
+                        AICLI_LOG(Core, Error, << "Failed to open remote settings, downloaded temp remote settings not trusted.");
+                        THROW_HR(APPINSTALLER_CLI_ERROR_REMOTE_SETTINGS_NOT_TRUSTED);
+                    }
+
+                    // Move and lock the newly downloaded settings file.
+                    std::filesystem::rename(tempSettingsPath, m_settingsFile);
+                    if (!LockSettingsFile(m_settingsFile, m_fileLock))
+                    {
+                        AICLI_LOG(Core, Error, << "Failed to open remote settings, downloaded remote settings not trusted.");
+                        THROW_HR(APPINSTALLER_CLI_ERROR_REMOTE_SETTINGS_NOT_TRUSTED);
+                    }
+                }
+
+                AICLI_LOG(Repo, Info, << "Opened trusted remote settings.");
+            }
+
+            std::unique_ptr<std::istream> Get() override
+            {
+                if (std::filesystem::exists(m_settingsFile))
+                {
+                    auto result = std::make_unique<std::ifstream>(m_settingsFile, std::ios::binary);
+                    THROW_LAST_ERROR_IF(result->fail());
+                    return result;
+                }
+                else
+                {
+                    return {};
+                }
+            }
+
+            bool Set(std::string_view) override
+            {
+                THROW_HR(E_UNEXPECTED);
+            }
+
+            void Remove() override
+            {
+                THROW_HR(E_UNEXPECTED);
+            }
+
+            std::filesystem::path PathTo() override
+            {
+                THROW_HR(E_UNEXPECTED);
+            }
+
+        private:
+            // Try to use cached copy of the settings file if applicable.
+            // Returns true if the settings file is trusted, up to date and successfully locked. Otherwise false.
+            bool TryLockCachedSettingsFile()
+            {
+                if (!std::filesystem::exists(m_settingsFile))
+                {
+                    AICLI_LOG(Core, Info, << "Cached remote settings file does not exist.");
+                    return false;
+                }
+
+                if (!LockSettingsFile(m_settingsFile, m_fileLock))
+                {
+                    return false;
+                }
+
+                Msix::MsixInfo remoteInfo{ s_RemoteSettings_Url };
+                if (remoteInfo.IsNewerThan(m_settingsFile))
+                {
+                    AICLI_LOG(Core, Info, << "Cached remote settings file is not up to date.");
+                    return false;
+                }
+
+                return true;
+            }
+
+            bool LockSettingsFile(const std::filesystem::path& settingsFile, std::unique_ptr<Msix::WriteLockedMsixFile>& lock)
+            {
+                lock = std::make_unique<Msix::WriteLockedMsixFile>(settingsFile);

```

Comment:
Is there a built-in wait for this lock creation, or could we be creating spurious failures due to contention over locking this file to read it?  Given that you are already planning on reading the entire thing in to memory, why not just read the file into memory without a lock.